### PR TITLE
Add sybil examples to more files

### PR DESCRIPTION
**Pull request by joaander**
## Description

Add sybil examples to all *.py files in the `hoomd` source directory. Also reword some documentation for clarity and reduce the toctree depth.

## Motivation and context

Provide users with code examples that are tested and known to execute without errors.

## How has this been tested?

I executed sybil tests locally (serial CPU build).

## Change log

No additional change log is needed over that in #1574.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/write/gsd.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ class GSD(Writer):
 
     Args:
         trigger (hoomd.trigger.trigger_like): Select the timesteps to write.
-        filename (str): File name to write.
+        filename (`str`): File name to write.
```
Comment (b-butler) [inline hoomd/write/gsd.py]: Given the explicit conversion to a string, we now would accept `PurePath` objects as well.

File: hoomd/simulation.py [hunk 2]
```diff
@@ -187,6 +215,12 @@ def create_state_from_gsd(self,
             automatically selected direction. The default value of ``(None,
             None, None)`` will automatically select the number of domains in all
             directions.
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            simulation.create_state_from_gsd(filename = path / 'file.gsd')
```
Comment (b-butler) [inline hoomd/simulation.py]: I think that adding a variable `filename` to the invisible code block would be more understandable for the new Python user as `Path` objects are not well known from my estimation.

File: hoomd/md/update.py [hunk 3]
```diff
@@ -220,7 +220,7 @@ def _attach_hook(self):
 
     @log(category="scalar", requires_run=True)
     def summed_exchanged_momentum(self):
-        R"""Returned the summed up exchanged velocity of the full simulation."""
+        """Returned the summed up exchanged velocity of the full simulation."""
```
Comment (b-butler) [inline hoomd/md/update.py]:
```suggestion
        """Returns the summed up exchanged velocity of the full simulation."""
```

File: hoomd/logging.py [hunk 4]
```diff
@@ -657,14 +625,20 @@ class they come from. For instance, the `hoomd.md.pair.LJ` class has a
...
         only_default (`bool`, optional): Whether to log only quantities that are
             logged by default. Defaults to ``True``. Non-default quantities
             are typically measures of operation performance rather than
             information about simulation state.
+
+    .. rubric:: Example:
+
+    .. code-block:: python
+
+        logger = hoomd.logging.Logger()
```
Comment (b-butler) [inline hoomd/logging.py:641]: See my PR #1588 where I added example logger constructions. We should choose one to keep.

**Review by b-butler — COMMENTED** [review]
The PR looks good overall I just have a few small questions/comments.

File: hoomd/write/gsd.py (hunk 1 above)
Comment (joaander) [reply hoomd/write/gsd.py]: Do most HOOMD users know what a `PurePath` object is? We also accept any time that will convert to a filename with `str()` - e.g. `filename=12345`. What is the universal way of documenting that we accept any type that will convert to a string?

File: hoomd/simulation.py (hunk 2 above)
Comment (joaander) [reply hoomd/simulation.py]: I'm not certain that `filename=filename` would be any less confusing to novice users.

File: hoomd/logging.py (hunk 4 above)
Comment (joaander) [reply hoomd/logging.py:641]: If you think the additional examples are helpful, we can keep them. I have been avoiding code comments in examples. If additional explanation or a header is needed, put that outside the code block. For example: https://github.com/glotzerlab/hoomd-blue/blob/4e30e11957421a349da1371a6980c12497d058cf/hoomd/md/methods/methods.py#L107-L123

File: hoomd/box.py [hunk 5]
```diff
@@ -176,15 +160,19 @@ class Box:
     .. rubric:: Factory Methods
 
     `Box` has factory methods to enable easier creation of boxes: `cube`,
-    `square`, `from_matrix`, and `from_box`. See the method documentation for
-    usage.
+    `square`, `from_matrix`, and `from_box`. See the each methods documentation
```
Comment (SchoeniPhlippsn) [inline hoomd/box.py]:
```suggestion
    `square`, `from_matrix`, and `from_box`. See the methods documentation
```

File: hoomd/box.py [hunk 6]
```diff
@@ -455,12 +546,17 @@ def to_matrix(self):
         """(3, 3) `numpy.ndarray` `float`: The upper triangular matrix that \
         defines the box.
 
-        .. code-block:: python
+        .. code-block::
```
Comment (SchoeniPhlippsn) [inline hoomd/box.py:554]: In line 227 you removed `.. code-block::` completely. I would make it consistent throughout the whole file.

File: hoomd/logging.py [hunk 7]
```diff
@@ -681,13 +655,13 @@ def categories(self):
 
     @property
     def string_categories(self):
-        """`list` of `str`: A list of the string names of the allowed \
+        """list[str]: A list of the string names of the allowed \
```
Comment (SchoeniPhlippsn) [inline hoomd/logging.py:658]:
```suggestion
        """`list` [`str`]: A list of the string names of the allowed \
```

File: hoomd/simulation.py (hunk 2 above)
Comment (SchoeniPhlippsn) [reply hoomd/simulation.py]: I agree that `filename = filename` might not be the best choice as well. Maybe we can rename `path` to something more specific such as `path_to_input_file`.

File: hoomd/state.py [hunk 8]
```diff
@@ -348,32 +363,74 @@ def set_snapshot(self, snapshot):
 
     @property
     def particle_types(self):
-        """list[str]: List of all particle types in the simulation state."""
+        """list[str]: List of all particle types in the simulation state.
```
Comment (SchoeniPhlippsn) [inline hoomd/state.py:366]:
```suggestion
        """`list` [`str`]: List of all particle types in the simulation state.
```
Same for all the following

**Review by SchoeniPhlippsn — COMMENTED** [review]
PR looks good. Only some minor comments.

File: hoomd/simulation.py (hunk 2 above)
Comment (b-butler) [reply hoomd/simulation.py]: It isn't so much the name, but the `Path("/path/to/tmp/dir") / "file.gsd")` syntax that I thought might be advanced.

File: hoomd/logging.py (hunk 7 above)
Comment (joaander) [reply hoomd/logging.py:658]: These are parsed correctly as is. The napoleon engine in sphinx understands these type annotations and expands `list` and `str` to links. We need explicit links only in cases where we add additional text to the type e.g. "`list` [`str`], optional".

File: hoomd/box.py (hunk 6 above)
Comment (joaander) [reply hoomd/box.py:554]: I made this change intentionally. This block is not a code example, it is a monospace font formatted matrix. I removed the sphinx `.. code-block::` directive and used the restructured text literal block instead: https://docutils.sourceforge.io/0.4/docs/ref/rst/restructuredtext.html#literal-blocks

All visible code examples that execute in sybil must be preceded by the sphinx block `.. code-block:: python`: https://sybil.readthedocs.io/en/latest/rest.html#python - other `.. code-block::` blocks are ignored by sybil.

File: hoomd/logging.py (hunk 4 above)
Comment (joaander) [reply hoomd/logging.py:641]: `logger` is needed to run the later examples in the file. We will add more examples (if any) during merge conflict resolution.

File: hoomd/box.py (hunk 6 above)
Comment (joaander) [reply hoomd/box.py:554]: I added a math formatted matrix.

**Review by b-butler — APPROVED** [review]
LGTM.

**Review by SchoeniPhlippsn — APPROVED** [review]

**Pull request merged**